### ISSUE TITLE
Exception while getting package (but the package does exist)

### ISSUE THREAD

**Issue by sztomi** [conan-io/conan#5055]
**Conan version**: 1.14.4, 1.4.4
**OS**: macOS 10.14.4

---

When installing a particular package on macOS, I get the following exception:

```
plex-media-server/1.15.6.1019-30040f981@plex/stable: Retrieving package abcc97af5961accce2d0377e72ac64b93e2c804e from remote 'experimental' 
Downloading conanmanifest.txt

Downloading conaninfo.txt

Downloading conan_package.tgz

....
plex-media-server/1.15.6.1019-30040f981@plex/stable: ERROR: Exception while getting package: abcc97af5961accce2d0377e72ac64b93e2c804e
plex-media-server/1.15.6.1019-30040f981@plex/stable: ERROR: Exception: <class 'FileNotFoundError'> [Errno 2] No such file or directory
plex-media-server/1.15.6.1019-30040f981@plex/stable: WARN: Trying to remove package folder: /Users/tamas/.conan_plex/.conan/data/plex-media-server/1.15.6.1019-30040f981/plex/stable/package/abcc97af5961accce2d0377e72ac64b93e2c804e
Traceback (most recent call last):
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/model/ref.py", line 111, in loads
    _, name, version, user, channel, revision, _ = ConanFileReference.sep_pattern.split(text)
ValueError: not enough values to unpack (expected 7, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/command.py", line 377, in install
    ref = ConanFileReference.loads(args.path_or_reference)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/model/ref.py", line 114, in loads
    "OpenCV/1.0.6@user/stable" % text)
conans.errors.ConanException: Wrong package recipe reference /Users/tamas/work/desktop
Write something like OpenCV/1.0.6@user/stable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/command.py", line 1579, in run
    method(args[0][1:])
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/command.py", line 390, in install
    install_folder=args.install_folder)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/conan_api.py", line 93, in wrapper
    return f(*args, **kwargs)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/conan_api.py", line 580, in install
    no_imports=no_imports)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/manager.py", line 80, in install
    installer.install(deps_graph, keep_build)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/installer.py", line 302, in install
    self._build(nodes_by_level, keep_build, root_node, graph_info)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/installer.py", line 324, in _build
    self._handle_node_cache(node, keep_build, processed_package_refs)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/installer.py", line 395, in _handle_node_cache
    self._recorder)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/remote_manager.py", line 148, in get_package
    touch_folder(dest_folder)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/util/files.py", line 76, in touch_folder
    os.utime(os.path.join(dirname, fname), None)
FileNotFoundError: [Errno 2] No such file or directory

ERROR: [Errno 2] No such file or directory
```

At first I thought this was just a case of a missing build, but that doesn't seem to be the case. The error is different and when I look for the package `abcc97af5961accce2d0377e72ac64b93e2c804e` in Artifactory, I can find it:

![image](https://user-images.githubusercontent.com/240594/56914821-605f6480-6ab5-11e9-97fe-c71fbda363f5.png)

I originally encountered this issue on 1.4.4, but upgraded quickly to rule out an incompatibility with the revisions migration in Artifactory. The issue reproduces 100% with the same package on different machines and different tries.

What can I do to troubleshoot this? 

---

To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by sztomi** [comment]
The contents of the conan_package.tgz seem to be fine. I tried with a different build (newer version), but that also reproduces the issue.

**Comment by jgsogo** [comment]
Hi @sztomi,

There is an error in the log:

```
conans.errors.ConanException: Wrong package recipe reference /Users/tamas/work/desktop
```

which says that the reference Conan is trying to parse is `/Users/tamas/work/desktop` which is not a valid reference, it looks like a path from another computer where the package was created, but I have no idea how that path gets there... 

It's good to know that the error is reproducible, we need to investigate it and we need to give a meaningful error message... but we need to be able to reproduce it or get more information about it. I don't know if you can share with us a minimum reproducible example, or maybe you can share with us the package files and the recipe (info@conan.io) and we will try to reproduce it here.

Thanks

**Comment by sztomi** [comment]
@jgsogo that error is completely misleading, and I think I reported that before (and in fact, the path is valid). It only displays that error when another exception happens. The real backtrace is the one that contains the `touch_folder` call. 

That said, I managed to figure out the issue:

1. Package contains a dead symlink (i.e. it points to an absolute path on the CI slave that built it)
2. Conan downloads package, calls `touch_folder`
3. `touch_folder` calls `os.utime` for each file, which raises `FileNotFoundError` when it reaches the dead symlink

--- 

Even though the root cause of the issue is in the package, I'd like to re-purpose this issue for improving the error reporting in this case. There were several obstacles that even required editing conan source to figure this out. Namely:

1. To reiterate: the following parts of the reported error are irrelevant (and in my experience, this is always reported and almost always irrelevant & confusing):

```
Traceback (most recent call last):
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/model/ref.py", line 111, in loads
    _, name, version, user, channel, revision, _ = ConanFileReference.sep_pattern.split(text)
ValueError: not enough values to unpack (expected 7, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/client/command.py", line 377, in install
    ref = ConanFileReference.loads(args.path_or_reference)
  File "/Users/tamas/.conan_plex/lib/conan-1.14.4-darwin.pyz/conans/model/ref.py", line 114, in loads
    "OpenCV/1.0.6@user/stable" % text)
conans.errors.ConanException: Wrong package recipe reference /Users/tamas/work/desktop
Write something like OpenCV/1.0.6@user/stable

During handling of the above exception, another exception occurred:
```

2. `RemoteManager` unconditionally removes the package directory after the error is caught: https://github.com/conan-io/conan/blob/157f0cbed244ce2065a0a5f8d4713ce5005ecff8/conans/client/remote_manager.py#L161

This makes it impossible to inspect the files afterwards. I think an environment variable could be used to make this optional.

3. `touch_folder` should catch FileNotFoundError and report the missing filename, because without that, it's impossible to tell what the missing file is. It seems like this differs a bit between 1.14.4 and master, because on master it swallows the exception: https://github.com/conan-io/conan/blame/157f0cbed244ce2065a0a5f8d4713ce5005ecff8/conans/util/files.py#L78 - not sure if that's a good idea unconditionally, without logging anything.

**Comment by jgsogo** [comment]
I think that the first place where we should notice about this broken symlink is during _untar_, there we should warn the user about files not in the package (https://github.com/conan-io/conan/issues/4966) and symlinks, probably we should also log (3) as you suggest (although here there shouldn't be any new file).

About (2) I think that we should leave the folder there (marked as `dirty` so it will be removed in the next run)

**Comment by lasote** [comment]
Closed by #5065

**Comment by lasote** [comment]
To be released at 1.14.5 asap
